rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -1,6 +1,11 @@
+local json = require("cjson")
+
 -- this is the Lua representation of Configuration struct in internal/ingress/types.go
 local configuration_data = ngx.shared.configuration_data
```

Comment:
We're attempting to make configuration_data's structure mirror this: https://github.com/kubernetes/ingress-nginx/blob/master/internal/ingress/types.go#L52

I feel like that breaks in this PR. Maybe we want a `servers` key within configuration_data? 

On the same vein; in addition, in your corresponding controller PR - we already have an [`ingress.Server` type](https://github.com/kubernetes/ingress-nginx/blob/master/internal/ingress/types.go#L137), which we could use instead of your new Certificate type in your [`configureCerts` method](https://github.com/Shopify/ingress/pull/58/files#diff-cde3fffe2425ad7efaa8add1d05ae2c0R829). What do you think?

E.g. 
````
servers := make([]*ingress.Server, len(pcfg.Servers))

for i, server := range pcfg.Servers {
  luaServer := &ingress.Server{
      ...
   }

Reply:
Good point, I agree. I will update the PRs.

Reply:
I talked to Elvin who wrote this comment and he said that this didn't have to mirror the Configuration struct in internal/ingress/types.go perfectly. I reverted the commit that used `servers` as a key, so we don't have to JSON decode the `servers` data and iterate through each server for every request.